### PR TITLE
Convert navigation menu item definition to managed entities

### PR DESCRIPTION
**Pull request by jensschuppe**

### CODE REVIEW

File: managed/Navigation__eck_entities.mgd.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        'icon' => 'crm-i fa-cubes',
+        'permission' => [
+          'access CiviCRM',
+        ],
+        'permission_operator' => 'OR',
+        'parent_id' => NULL,
+        'is_active' => TRUE,
+        'has_separator' => 0,
+      ],
+      'match' => ['name', 'parent_id'],
```
Comment (colemanw) [inline managed/Navigation__eck_entities.mgd.php:42]:
```suggestion
      'match' => ['name', 'parent_id', 'domain_id'],
```

**Comment by colemanw** [comment]
This looks good @jensschuppe. Only change is to add 'domain_id' to the "match" lines, and delete the 2 lines `'domain_id' => 'current_domain',` (those will be auto-managed by CiviCRM).

**Comment by jensschuppe** [comment]
Thanks @colemanw! So `Navigation` entities are always only unique by `name`, `parent_id` and `domain_id`?

**Comment by jensschuppe** [comment]
… and do you know why the navigation items have not been available for moving/editing in the navigation menu admin UI when they weren't managed entities, but defined by civix methods (as described in #96)?

**Comment by colemanw** [comment]
@jensschuppe yes I've written up a bunch on that topic: https://lab.civicrm.org/dev/core/-/issues/4711

**Pull request merged**